src/main/java/com/ss/training/lms/controller/AdminGenreController.java

```diff
@@ -44,77 +53,94 @@
 		return new ResponseEntity<Genre>(genre, status);
 	}
 
-	@RequestMapping(path="/lms/admin/genres", 
-					produces = {
+	/**
+	 * 
+	 * @return
+	 */
+	@GetMapping(produces = {
 						MediaType.APPLICATION_XML_VALUE,
 						MediaType.APPLICATION_JSON_VALUE
 					})
-	public ResponseEntity<List<Genre>> getAllGenres() {
+	public ResponseEntity<List<Genre>> readGenres() {
 		List<Genre> genres = null;
 		HttpStatus status = HttpStatus.OK;
         try {
 			genres= genreService.readAllGenres();
 			if (genres == null) {
-				status = HttpStatus.NOT_FOUND;
+				status = HttpStatus.NO_CONTENT;
 			}
 		} catch (ClassNotFoundException | SQLException e) {
 			status = HttpStatus.INTERNAL_SERVER_ERROR;
 		}
         return new ResponseEntity<List<Genre>>(genres, status);
 	}
 
-	
-	@RequestMapping(path="/lms/admin/genre",
-					method = RequestMethod.POST,
-					consumes = {
+	/**
+	 * 
+	 * @param genre
+	 * @return
+	 */
+	@PostMapping(consumes = {
 						MediaType.APPLICATION_JSON_VALUE,
 						MediaType.APPLICATION_XML_VALUE
 					})
-	public ResponseEntity<Genre> addGenre(@RequestBody Genre genre) {
-		if(genre.getGenreName() == null)
+	public ResponseEntity<Genre> createGenre(@RequestBody Genre genre) {
+		HttpStatus status = HttpStatus.BAD_REQUEST;
+		if(genre == null || genre.getGenreName().length() > 45)
```

Comment:
This will hit an exception if the genre isn't null but its name is. To avoid this, the second condition can be replaced with `(genre.getGenreName() != null && genre.getGenreName().length() > 45)`.